src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -111,6 +112,84 @@ bool TryParseUri(string u, out Uri uri)
 			}
 		}
 
+		protected double ParseMetricsInterval(ConfigurationKeyValue kv)
+		{
+			string value;
+			if (kv == null || string.IsNullOrWhiteSpace(kv.Value))
+			{
+				value = ConfigConsts.DefaultValues.MetricsInterval;
+			}
+			else
+			{
+				value = kv.Value;
+			}
+
+			if (! TryParseTimeInterval(value, out var valueInMilliseconds))
+			{
+				Logger?.Error()
+					?.Log("Failed to parse provided metrics interval `{ProvidedMetricsInterval}' - " +
+						"using default: {DefaultMetricsInterval}",
+						value,
+						ConfigConsts.DefaultValues.MetricsInterval);
+				return ConfigConsts.DefaultValues.MetricsIntervalInMilliseconds;
+			}
+
+			// ReSharper disable once CompareOfFloatsByEqualityOperator - we compare to exactly zero here
+			if (valueInMilliseconds == 0)
+				return valueInMilliseconds;
+
+			if (valueInMilliseconds < 0)
+			{
+				Logger?.Error()
+					?.Log("Provided metrics interval `{ProvidedMetricsInterval}' is negative - " +
+						"using default: {DefaultMetricsInterval}",
+						value,
+						ConfigConsts.DefaultValues.MetricsInterval);
+				return ConfigConsts.DefaultValues.MetricsIntervalInMilliseconds;
+			}
+
+			if (valueInMilliseconds < ConfigConsts.Constraints.MinMetricsIntervalInMillisecond)
+			{
+				Logger?.Error()
+					?.Log("Provided metrics interval `{ProvidedMetricsInterval}' is smaller than allowed minimum: {MinProvidedMetricsInterval}ms - " +
+						"metrics collection will be disabled",
+						value,
+						ConfigConsts.Constraints.MinMetricsIntervalInMillisecond);
+				return 0;
+			}
+
+			return valueInMilliseconds;
+		}
+
+		private bool TryParseTimeInterval(String valueAsString, out double valueInMilliseconds)
```

Comment:
@SergeyKleyman One small nit to this one, I think pattern matching is more readable than the lots of if statements in the method. But I'm also ok with this.